enhancements/external-secrets-operator/external-secrets-operator.md

```diff
@@ -0,0 +1,549 @@
+through the status sub-resource of the new custom resource. `external-secrets-operator` will make use of static manifest templates derived from 
+the helm charts provided by the `external-secrets` project for creating the resources required for successfully deploying `external-secrets`. 
+Please refer to the `Implementation Details/Notes/Constraints` section for more details. external-secrets-operator watches the new CR 
+`externalsecrets.operator.openshift.io` and deploys the resources required for `external-secrets`.
+
+Each of the resources created for external-secrets deployment will have the below set of labels added, along with any additional labels configured by the user.
+- `app: external-secrets-operator`
+- `app.kubernetes.io/name: external-secrets`
+- `app.kubernetes.io/instance: external-secrets`
+- `app.kubernetes.io/version: "v0.14.0"`
+- `app.kubernetes.io/managed-by: external-secrets-operator`
+- `app.kubernetes.io/part-of: external-secrets-operator`
+
+Refer below links for more information on the labels used
+- [Guidelines for Labels and Annotations for OpenShift applications](https://github.com/redhat-developer/app-labels/blob/master/labels-annotation-for-openshift.adoc)
+- [Well-Known Labels, Annotations and Taints](https://kubernetes.io/docs/reference/labels-annotations-taints/)
+
+`externalsecrets.operator.openshift.io` CR object is a cluster-scoped object and is enforced in CRD to have the name as cluster.
+
+Configurations made available in the spec of `externalsecrets.operator.openshift.io` CR are passed as command line arguments to `external-secrets`
+and updating such configurations would cause the new rollout of the external-secrets agent deployment, which means a new pod will be created 
+and old pod will terminate resulting in external-secrets restart.
+
+When the user deletes `externalsecrets.operator.openshift.io` CR object, external-secrets-operator will stop managing all the resources 
+created for installing external-secrets and the user will have to manually clean up the resources. Please refer to the 
+`Operational Aspects of API Extensions` section for command to list all the resources.
+
+`externalsecrets.operator.openshift.io` CR status sub-resource will be used for updating the status of the external-secrets installation, 
+any error encountered while creating the required resources or the reconciling the state.
+
+`externalsecretsoperator.operator.openshift.io` is another CR object, which is made available for configuring global options and to 
+enable optional or TechPreview features. The CR object will be created during the operator installation with default values and user can
+modify or update the CR as required. 
+
+A fork of upstream [external-secrets](https://github.com/external-secrets/external-secrets) will be created [downstream](https://github.com/openshift/external-secrets) for better version management.
+
+### Workflow Description
+
+- Installation of external-secrets.
```

Comment:
qq: should we include what `users` would have permissions to install/uninstall etc?

Reply:
suggestion incorporated.